dashboards-observability/public/components/common/metrics_listener.tsx

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { HttpStart } from '../../../../../src/core/public';
+import { OBSERVABILITY_BASE } from '../../../common/constants/shared';
+
+interface MetricsListenerProps {
+  http: HttpStart;
+}
+
+export const MetricsListener: React.FC<MetricsListenerProps> = (props) => {
+  const incrementCountMetric = (element?: string | null) => {
+    if (!element) return;
+    props.http.post(`${OBSERVABILITY_BASE}/stats`, {
+      body: JSON.stringify({ element }),
+    });
+  };
+
+  const onClick: React.MouseEventHandler<HTMLDivElement> = (e) => {
+    incrementCountMetric(
+      (e.target as HTMLElement | null)
+        ?.closest('[data-click-metric-element]')
+        ?.getAttribute('data-click-metric-element')
```

Comment:
Where is this coming from - is the idea that we label buttons with what they are doing? I see it's being added to some test snapshots, but only for trace analytics refresh? Will it be added for different other operations in the future?

Reply:
Are we going to be sending these stats anywhere else other than the local index? Any place where we can get the aggregated stats?

Reply:
@derek-ho See description. This PR only adds the framework and some basic examples.

> Why can't you just have it handled when we send the request?

Instrumenting metrics in each element's onClick is tedious comparing to adding an attribute.

> Are we going to be sending these stats anywhere else other than the local index? Any place where we can get the aggregated stats?

Please see description for the "To get metrics" part.